code/game/gamemodes/cm_process.dm

```diff
@@ -170,6 +170,41 @@ dat += " You failed to evacuate \the [MAIN_SHIP_NAME]"
 		to_chat(world, dat)
 
 
+/datum/game_mode/proc/declare_completion_announce_round_stats()
+	set waitfor = 0
+	sleep(140)
+	for(var/mob/player in player_list)
+		if(player?.client?.prefs?.toggles_chat & CHAT_STATISTICS)
+			to_chat(player, "<span class='round_body'>The end of round statistics are:</span>")
+			to_chat(player, " ")//gotta get that formatting on.
+			to_chat(player, "There were [round_statistics.total_bullets_fired] total bullets fired.")
```

Comment:
You should build all this text first before sending it to each player rather than rebuilding it every time.

Reply:
```
var/list/dat = list({"<span class='round_body'>The end of round statistics are:</span><br>
	<br>There were [round_statistics.total_bullets_fired] total bullets fired.
	<br>[round_statistics.total_bullet_hits_on_marines] bullets managed to hit marines. For a [(round_statistics.total_bullet_hits_on_marines / max(round_statistics.total_bullets_fired, 1)) * 100]% friendly fire rate!"})
if(round_statistics.total_bullet_hits_on_xenos)
	dat += "[round_statistics.total_bullet_hits_on_xenos] bullets managed to hit xenomorphs. For a [(round_statistics.total_bullet_hits_on_xenos / max(round_statistics.total_bullets_fired, 1)) * 100]% accuracy total!"
if(round_statistics.grenades_thrown)
	dat += "[round_statistics.grenades_thrown] total grenades exploding."
else
	dat += "No grenades exploded."
if(round_statistics.now_pregnant)
	dat += "[round_statistics.now_pregnant] people infected among which [round_statistics.total_larva_burst] burst. For a [(round_statistics.total_larva_burst / max(round_statistics.now_pregnant, 1)) * 100]% successful delivery rate!"
if(round_statistics.queen_screech)
	dat += "[round_statistics.queen_screech] Queen screeches."
if(round_statistics.ravager_ravage_victims)
	dat += "[round_statistics.ravager_ravage_victims] ravaged victims. Damn, Ravagers!"
if(round_statistics.warrior_limb_rips)
	dat += "[round_statistics.warrior_limb_rips] limbs ripped off by Warriors."
if(round_statistics.crusher_stomp_victims)
	dat += "[round_statistics.crusher_stomp_victims] people stomped by crushers."
if(round_statistics.praetorian_spray_direct_hits)
	dat += "[round_statistics.praetorian_spray_direct_hits] people hit directly by Praetorian acid spray."
if(round_statistics.weeds_planted)
	dat += "[round_statistics.weeds_planted] weed nodes planted."
if(round_statistics.weeds_destroyed)
	dat += "[round_statistics.weeds_destroyed] weed tiles removed."
if(round_statistics.carrier_traps)
	dat += "[round_statistics.carrier_traps] hidey holes for huggers were made."
var/output = jointext(dat, "<br>"
for(var/mob/player in player_list)
	if(player?.client?.prefs?.toggles_chat & CHAT_STATISTICS)
		to_chat(player, output)
```
This is faster (not that it matters much here) as it avoids combining strings together until the end by joining a list of them.

Reply:
Can you do it as a list, as here suggested? Also, conflicts to be solved.

Reply:
done and done,